File: filt.cs
using System;
using System.Linq;

public class Filt {
    public static void Main(string[] args) {
        int[] d = {3, -1, 4, -2, 5, -3, 6, -4};
        int[] d2 = filt(d);
        foreach(var v in d2) {
            Console.WriteLine(v);
        }
    }

    public static int[] filt(int[] arr) {
        return arr.Where(x => x < 0).ToArray();
    }
}

File: filt.swift
import Foundation

func filt<T: BinaryInteger>(arr: [T]) -> [T] {
    return arr.filter { $0 < 0 }
}

let d = [3, -1, 4, -2, 5, -3, 6, -4]
let d2 = filt(arr: d)

for i in d2 {
    print(i)
}

